fix: remove intro markdown and handle all intro information from intro.json

jovicon opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #41785

<!-- Feel free to add any additional description of changes below this line -->
Hi, I've been studying the code because is my very first time working with gatsby and to contribute on FreecodeCamp.
I created this PR to upload my advances and doubts, also I think this a very huge change and I want to do it well.


What does this first commit have? 
install [gatsby-transformer-json](https://www.gatsbyjs.com/plugins/gatsby-transformer-json/) to handle json data.
Initially introTests.Json will evolve in intro.json ( using like pivot to test gatsby render right now )
adding allIntroTestsJson to Graphql structure

Doubt:
- The data that we will use from file intro.json
We have to add these fields on allIntroTestsJson: slug, nodeIdentity, component, block, superBlock, htmlAst, excerpt.
Where can I get them from?

jovicon committed:
Defining Object structure to Intro.json file

ojeytonwilliams commented:
> We have to add these fields on allIntroTestsJson: slug, nodeIdentity, component, block, superBlock, htmlAst, excerpt.

I don't believe that's necessary any more.  All we need to do now is supply `createSuperBlockIntroPages` with a `slug` and `superBlock`:

https://github.com/freeCodeCamp/freeCodeCamp/blob/2f736f7f8ef01b425c3f87c3c391a814671e8d4a/client/utils/gatsby/challengePageCreator.js#L99-L113

then https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/templates/Introduction/SuperBlockIntro.js should take care of everything.

Also, it might be easier just to import the JSON directly, rather than sourcing with Gatsby and querying it, since it's only used once during page creation.

jovicon commented:
Thanks @ojeytonwilliams I will check what you told me.

raisedadead commented:
Hi @jovicon how is this coming along? Let us know if there is anything we can do to help in case you are stuck.

jovicon commented:
Hi @raisedadead, sorry for taking to long. 🙏
@ojeytonwilliams thanks for your help. I finally solved this issue today.
I will push tomorrow because I have to make some code cleaning.

Again, sorry for taking to long.

jovicon committed:
new changes on markdown removes files task
* deleted: Almost all Markdown files
* modified:
	* propTypes: to add new json type
	* certAndProjectMap: to match title with front end
	* Superblock and Intro: change Graphql query
* new files: intro.json on learn folder

jovicon committed:
remove: uniq from gatsby-node.js

jovicon commented:
Hi @ojeytonwilliams, how are you?
Thanks for your help, your advice cleared me a lot of things about how to move forward.

I have a final question, the code is working fine.
But when I delete the last MD folder everything fall away.

* This is the last MD folder: `pages/learn/apis-and-microservices`.
* This is the code that I think is giving me problems. https://github.com/freeCodeCamp/freeCodeCamp/blob/2f736f7f8ef01b425c3f87c3c391a814671e8d4a/client/gatsby-node.js#L29-L38

Do you know what it could be?

ojeytonwilliams commented on client/src/templates/Introduction/Intro.js:
```diff
@@ -83,7 +83,7 @@ IntroductionPage.propTypes = propTypes;
 export const query = graphql`
-  query IntroPageBySlug($slug: String!, $block: String!) {
+  query IntroPageBySlug($block: String!) {
     markdownRemark(fields: { slug: { eq: $slug } }) {
       frontmatter {
         block
```
```suggestion
    markdownRemark(fields: { slug: { eq: $slug } }) {
      frontmatter {
        block
```
As I mentioned above, these shouldn't be necessary once the markdown files are gone.

ojeytonwilliams commented on package.json:
```diff
@@ -100,7 +100,8 @@
   "dependencies": {
-    "dotenv": "8.2.0"
+    "dotenv": "8.2.0",
+    "gatsby-transformer-json": "^3.3.0"
```
We have a, somewhat complicated, setup in which our Docker images are created: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/server.Dockerfile#L16 so could you move this to the client dir?  Otherwise it will become part of the api image.

ojeytonwilliams commented on client/gatsby-node.js:
```diff
@@ -36,6 +35,13 @@ exports.onCreateNode = function onCreateNode({ node, actions, getNode }) {
       createNodeField({ node, name: 'component', value: component });
```
GitHub's UI doesn't let me put the comment in the right place, sadly, but we shouldn't be generating any MarkdownRemark nodes, so I think this can go.

ojeytonwilliams reviewed (changes requested):
Hi @jovicon I'm well, thanks for asking!  How're you?

So, before we dig into the mystery of why things fall apart when you delete the MD folder I have some general observations.  

Ideally we shouldn't be recreating anything that's already in the i18n folder.  This is for a couple of reasons - we should keep the code DRY and any text we source needs to be in all the languages.  With that in mind, I would recommend either sourcing the appropriate file from the i18n folder or, if that can't be done, strip out title and intro from client/src/pages/learn/intro.json since those will need to be handled by the i18n library.

The other thing is that, if we're ripping out the .md files, we can remove all reference to Markdown in the code. My suspicion is that that will allow us to delete the pages/learn/apis-and-microservices folder.

jovicon commented:
Thanks for your review @ojeytonwilliams 
I'll be working this weekend on this changes.

ojeytonwilliams commented:
Awesome. Take whatever time you need.

raisedadead commented:
Going to close this stale for now. @jovicon please feel free to reopen this or create a fresh PR when ready.
